sdk/src/cache/messages/data/scalar/set.rs

```diff
@@ -85,11 +87,21 @@ impl<K: IntoBytes, V: IntoBytes> MomentoRequest for SetRequest<K, V> {
             },
         )?;
 
-        let _ = cache_client.next_data_client().set(request).await?;
-        Ok(SetResponse {})
+        let response = cache_client
+            .next_data_client()
+            .set(request)
+            .await?
+            .into_inner();
+        match response.result() {
+            ECacheResult::Ok => Ok(SetResponse {}),
+            _ => Err(MomentoError::unknown_error(
+                "Set",
+                Some(format!("{:#?}", response)),
+            )),
```

Comment:
previously, the `set` command didn't check for errors like the other sdks do, so this was added to be more consistent